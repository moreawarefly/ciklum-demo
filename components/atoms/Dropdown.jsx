import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Label = styled.label`

`;

const Select = styled.select`

`;

function Dropdown({
  className,
  required,
  label,
  items,
  onChangeHandler,
}) {
  return (
    <Label className={className}>
      {!!label && `${label}: `}
      <Select
        required={required}
        onChange={onChangeHandler}
        defaultValue={undefined}
      >
        <option value=""> -- select -- </option>
        {items.map(item => (
          <option
            key={item.key}
            value={item.value}
          >
            {(item.label)}
          </option>
        ))};
      </Select>
    </Label>
  );
}

Dropdown.defaultProps = {
  className: '',
  required: false,
  label: null,
  items: [],
  onChangeHandler: () => {},
};

Dropdown.propTypes = {
  className: PropTypes.string,
  required: PropTypes.bool,
  label: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string,
    value: PropTypes.string,
    label: PropTypes.string,
  })),
  onChangeHandler: PropTypes.func,
};

export default Dropdown;
