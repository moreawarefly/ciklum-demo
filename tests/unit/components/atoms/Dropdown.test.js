import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from '../../../../components/atoms/Dropdown';

describe('Dropdown', () => {
  it('renders correctly w/o arguments', () => {
    const tree = renderer
      .create(<Dropdown />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with provided all attributes', () => {
    const mockList = [
      {
        key: 'key1',
        value: 'value1',
        label: 'label1',
      },
      {
        key: 'key2',
        value: 'value2',
        label: 'label2',
      },
    ];

    const tree = renderer
      .create(
        <Dropdown
          items={mockList}
          name="mockName"
          required
          label="mockLabel"
          onChangeHandler={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
