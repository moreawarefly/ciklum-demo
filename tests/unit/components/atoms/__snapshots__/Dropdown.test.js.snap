// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly w/o arguments 1`] = `
<label
  className="Dropdown__Label-sc-18glqnp-0 eUXFKX"
>
  <div
    className="Dropdown__SelectWrapper-sc-18glqnp-1 dymhwW"
    required={false}
  >
    <select
      className="Dropdown__Select-sc-18glqnp-2 jRkxKY"
      name=""
      onChange={[Function]}
      required={false}
    >
      <option
        value=""
      >
         -- select -- 
      </option>
      ;
    </select>
  </div>
</label>
`;

exports[`Dropdown renders correctly with provided all attributes 1`] = `
<label
  className="Dropdown__Label-sc-18glqnp-0 eUXFKX"
>
  mockLabel: 
  <div
    className="Dropdown__SelectWrapper-sc-18glqnp-1 loaLWB"
    required={true}
  >
    <select
      className="Dropdown__Select-sc-18glqnp-2 jRkxKY"
      name="mockName"
      onChange={[Function]}
      required={true}
    >
      <option
        value=""
      >
         -- select -- 
      </option>
      <option
        value="value1"
      >
        label1
      </option>
      <option
        value="value2"
      >
        label2
      </option>
      ;
    </select>
  </div>
</label>
`;
